platform/darwin/src/MGLFeature.mm

```diff
@@ -51,6 +58,56 @@ - (NSString *)description
             self.attributes.count ? self.attributes : @"none"];
 }
 
+- (id<MGLPointCluster>)cluster {
+    return self.cluster ? self : nil;
+}
+
+#pragma mark - MGLPointCluster
+
+- (BOOL)isCluster {
+    NSNumber *isCluster = MGL_OBJC_DYNAMIC_CAST(self.attributes[MGLPointClusterBoolKey], NSNumber);
+    return [isCluster boolValue];
+}
+
+- (uint32_t)clusterId {
+    NSNumber *clusterNumber = MGL_OBJC_DYNAMIC_CAST(self.attributes[MGLPointClusterIdKey], NSNumber);
+    
+    NSAssert(clusterNumber, @"Clusters should have a cluster_id");
+    if (!clusterNumber) {
+        return -1;
+    }
+
+    NSUInteger clusterId = [clusterNumber unsignedIntegerValue];
+
+    NSAssert(clusterId < (1UL << 32), @"Cluster IDs are 32bit");
+    NSAssert([self.identifier isKindOfClass:[NSNumber class]], @"The identifier should be an NSNumber");
+    NSAssert(clusterId == [self.identifier unsignedIntegerValue], @"The cluster id should match the feature's identifier.");
+    
+    return (uint32_t)clusterId;
+}
+
+- (NSUInteger)pointCount {
+    NSNumber *pointCount = MGL_OBJC_DYNAMIC_CAST(self.attributes[MGLPointClusterCountKey], NSNumber);
+    
+    NSAssert(pointCount, @"Clusters should have a point_count");
+    if (!pointCount) {
+        return -1;
+    }
+
+    return [pointCount unsignedIntegerValue];
+}
+
+- (NSString*)clusterAbbreviated {
```

Comment:
“Abbreviated” feels ungrammatical here and I’m unsure what this method is meant to do.

Reply:
It's really the *point count* represented in a text friendly form. I agree that the name is unpleasant, but it currently matches the underlying data & JS ("cluster_abbreviated"). How do you feel about deviating from that? If so, we could come up with something a lot clearer 🙂 

For that matter, we could address the mixed-metaphor of `leaves` vs `children` - though if we do that we'd probably want to do the same on Android. What do you think @tobrun ?

Reply:
Have removed the abbreviation property from latest rebase/push.